std/math.d

```diff
@@ -6279,12 +6279,26 @@ Unqual!(Largest!(F, G)) pow(F, G)(F x, G y) @nogc @trusted pure nothrow
         {
             // Result is real only if y is an integer
             // Check for a non-zero fractional part
-            if (y > -1.0 / real.epsilon && y < 1.0 / real.epsilon)
-            {
-                long w = cast(long)y;
-                if (w != y)
+            enum maxOdd = pow(2.0L, real.mant_dig) - 1.0L;
+            static if(maxOdd > ulong.max) {
+                // Generic method, for any FP type
+                if(floor(y) != y)
                     return sqrt(x); // Complex result -- create a NaN
-                if (w & 1) sign = -1.0;
+
+                const hy = ldexp(y, -1);
+                if(floor(hy) != hy)
+                    sign = -1.0;
+            } else {
```

Comment:
Actually, sorry to be a bother, but curly braces on the next line.  This format is used everywhere.
```
static if(...)
{
}
else
{
}
```